src/app/pages/question/question.component.ts

```diff
@@ -29,16 +31,24 @@ export class QuestionComponent implements OnInit, CanDeactivate<CanComponentDeac
   currentIndex$ = new BehaviorSubject<number>(0);
   currentQuestion$!: Observable<QuestionModel>;
   answers: string[] = new Array<string>(QUESTIONS_SIZE);
+  
   modalWindowState$ = new BehaviorSubject<boolean>(false);
   modalWindowChoice$ = new BehaviorSubject<boolean>(false);
+  modalWindowData$ = new BehaviorSubject<ModalWindowModel>({ page: '', title: '', text: '', link: '' });
 
   constructor(
     private activeRoute: ActivatedRoute,
     private router: Router,
     private storeService: StoreService,
+    private modalService: ModalWindowService
   ) {}
 
   canDeactivate(): Observable<boolean> {
+    if (this.currentIndex$.value == QUESTIONS_SIZE) {
+      this.modalWindowData$ = new BehaviorSubject<ModalWindowModel>(this.modalService.getData('Finish'));
+    }
+    const item = event?.target as HTMLElement;
```

Comment:
Where do we get this event from? I can't spot any related content

Reply:
I got that I need to avoid event on this, but when I wrote the code everything was good and I didn't have thoughts that it's wrong way to do like this.